server/src/test/java/org/elasticsearch/common/RoundingTests.java

```diff
@@ -232,6 +232,21 @@ public void testRandomTimeUnitRounding() {
         }
     }
 
+    /**
+     * This test chooses a date in the middle of the transition, so that we can test
+     * if the transition which is before the minLookup, but still should be applied
+     * is not skipped
+     */
+    public void testRoundingAroundDST() {
+        Rounding.DateTimeUnit unit = Rounding.DateTimeUnit.DAY_OF_MONTH;
+        ZoneId tz = ZoneId.of("Canada/Newfoundland");
+        long minLookup = 688618001000L; // 1991-10-28T02:46:41.527Z
+        long maxLookup = 688618001001L; // +1sec
+        // there is a Transition[Overlap at 1991-10-27T00:01-02:30 to -03:30] ”
```

Comment:
note this tricky 00:01. so if we convert: 
1991-10-28T02:46:41.527Z -> 1991-10-17 23:11:48 -03:30 Canada/Newfoundland
then we round to DAY_OF_MONTH
1991-10-27 00:00:00 -**02**:30 Canada/Newfoundland

Previously it would not work because we only included transition for timestamps 1unit before the minUtcMillis
//            long minLookup = minUtcMillis - 2*unit.extraLocalOffsetLookup();
So for 1991-10-28T02:46:41.527Z it would be 1991-10-**27T02**:46:41.527Z

which means we would miss the relevant transition 1991-10-**27T00**:01-02:30 to -03:30